examples/compound/compound.go

```diff
@@ -0,0 +1,274 @@
+	headers := make(chan *types.Header)
+	_, err = client.SubscribeNewHead(context.Background(), headers)
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to subscribe headers")
+	}
+
+	sub := c.CEtherLogsListener(client, logs)
+
+	for {
+		select {
+		case header := <-headers:
+			log.Debug().
+				Str("block", header.Number.String()).
+				Uint64("ts", header.Time).
+				Msg("header received")
+		case err = <-sub.Err():
+			log.Fatal().Err(err)
+		case evLog := <-logs:
+			msg, err := c.ProcessLogEvent(contractAbi, evLog)
+			if err != nil {
+				log.Error().Err(err).Msg("failed to process log event")
+			}
+			// Not sure what value needs to be passed as subject
+			err = c.ProduceMessage(namespace, evLog.Address.Hex(), msg)
+			if err != nil {
+				log.Error().Err(err).Msg("Kafka write proto")
+			}
+			// Commit Kafka Transaction
+			err = c.Producer.CommitTransaction(nil)
+			if err != nil {
+				log.Error().Err(err).Msg("Processor: Failed to commit transaction")
+
+				err = c.Producer.AbortTransaction(nil)
+				if err != nil {
+					log.Fatal().Err(err).Msg("")
+				}
+			}
+			log.Info().Interface("msg", msg).Msg("message delivered")
+			// Start a new transaction
+			err = c.BeginTransaction()
+			if err != nil {
+				log.Fatal().Err(err).Msg("")
+			}
+		case <-interrupt:
+			log.Info().Msg("interrupt")
+
+			// Cleanly close the connection by sending a close message and then
+			// waiting (with timeout) for the server to close the connection.
+			client.Close()
+			c.Producer.Close()
+			return
+		}
+	}
+
+}
+
+// CEtherLogsListener listens to all contracts that emit CEther related events.
+func (c *Connector) CEtherLogsListener(client *ethclient.Client, logs chan types.Log) ethereum.Subscription {
+	query := ethereum.FilterQuery{
+		Addresses: c.ContractAddresses,
+	}
+
+	sub, err := client.SubscribeFilterLogs(context.Background(), query, logs)
+	if err != nil {
+		log.Fatal().Err(err).Msg("subscribing CEther filter logs failed")
+	}
+
+	return sub
+}
+
+func (c *Connector) ProcessLogEvent(contractAbi abi.ABI, evLog types.Log) (proto.Message, error) {
+	// TODO: Add timestamp from block since logs don't include timestamp
+	ev, err := contractAbi.EventByID(evLog.Topics[0])
+	if err != nil {
+		log.Warn().Err(err).Msg("EventByID error, skipping")
+		return nil, err
+	}
+
+	if ev == nil {
+		log.Warn().Msg("ignore if event id isn't defined in a partial ABI")
+		return nil, err
+	}
+
+	var msg proto.Message
+
+	switch ev.Name {
+	case "Mint":
+		event := new(ctoken.CompoundMint)
+		if err := UnpackLog(contractAbi, event, ev.Name, evLog); err != nil {
+			log.Error().Err(err).Msg("Unpack Mint event error")
+			return nil, err
+		}
+		msg = &ctoken.Mint{
+			Ts:         timestamppb.Now(),
+			Block:      evLog.BlockNumber,
+			Idx:        uint64(evLog.Index),
+			Tx:         evLog.TxHash.Bytes(),
+			Minter:     event.Minter.Bytes(),
+			MintAmount: event.MintAmount.Bytes(),
+			MintTokens: event.MintTokens.Bytes(),
+		}
+
+		break
```

Comment:
I don't think we need `break` in golang. Also can directly `return &ctoken.Mint{}, nil`

Reply:
Removed `break`